src/Tests/dotnet-workload-install.Tests/GivenNetSdkManagedWorkloadInstall.cs

```diff
@@ -38,22 +38,34 @@ public void GivenManagedInstallItsInsallationUnitIsPacks()
             installer.GetInstallationUnit().Should().Be(InstallationUnit.Packs);
         }
 
-        [Fact]
-        public void GivenManagedInstallItCanGetFeatureBands()
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public void GivenManagedInstallItCanGetFeatureBands(bool writeRecords)
         {
             var versions = new string[] { "6.0.100", "6.0.300", "7.0.100" };
-            var (dotnetRoot, installer, _) = GetTestInstaller();
+            var (dotnetRoot, installer, _) = GetTestInstaller(identifier: writeRecords.ToString());
 
             // Write fake workloads
             foreach (var version in versions)
             {
                 var path = Path.Combine(dotnetRoot, "metadata", "workloads", version, "InstalledWorkloads");
                 Directory.CreateDirectory(path);
-                File.Create(Path.Combine(path, "6.0.100"));
+                if (writeRecords)
+                {
+                    File.Create(Path.Combine(path, "6.0.100"));
+                }
             }
 
             var featureBands = installer.GetFeatureBandsWithInstallationRecords();
-            featureBands.ShouldBeEquivalentTo(versions);
+            if (writeRecords)
+            {
+                featureBands.ShouldBeEquivalentTo(versions);
+            }
+            else
```

Comment:
nit: there should be less or prefer not branching (if-else) in test, or you could have bug it test code itself. In this case better to just split into 2 tests